contracts/sapphire/FlashLiquidator/FlashLiquidatorAaveV2.sol

```diff
@@ -0,0 +1,235 @@
+import {ISapphirePool} from "../SapphirePool/ISapphirePool.sol";
+import {SapphireTypes} from "../SapphireTypes.sol";
+
+/**
+ * @title FlashLiquidatorAaveV2
+ * @author ARCx
+ * @notice A basic flash loan based contracts that is used to liquidate borrowers of SapphireCoreV1
+ * contracts. It takes a flash loan to repay the user's debt, swaps the collateral to the
+ * repaid stablecoin, transfers the profit to the caller of the liquidate function, then repays
+ * the flash loan debt.
+ */
+contract FlashLiquidatorAaveV2 is FlashLoanReceiverBase, Ownable {
+
+    event ProfitReceiverSet(address indexed profitReceiver);
+
+    address public profitReceiver;
+
+    struct LiquidationVariables {
+        address vaultOwner;
+        SapphireTypes.ScoreProof[] proofs;
+        ISapphireCoreV1 core;
+        address collateralAsset;
+    }
+
+    ISwapRouter public immutable swapRouter;
+
+    constructor(
+        ILendingPoolAddressesProvider _provider,
+        ISwapRouter _router,
+        address _profitReceiver
+    )
+        FlashLoanReceiverBase(_provider)
+    {
+        swapRouter = _router;
+        profitReceiver = _profitReceiver;
+    }
+
+    function liquidate(
+        address _core,
+        address _owner,
+        SapphireTypes.ScoreProof memory _creditScoreProof
+    )
+        external
+    {
+        ISapphireCoreV1 core = ISapphireCoreV1(_core);
+
+        // Get a valid repayment asset
+        ISapphirePool pool = ISapphirePool(core.borrowPool());
+        IERC20Metadata repayAsset = IERC20Metadata(pool.getDepositAssets()[0]);
+        uint8 repayAssetDecimals = repayAsset.decimals();
+
+        // get user's current debt
+        SapphireTypes.Vault memory vault = core.vaults(_owner);
+        uint256 outstandingDebt = Math.roundUpMul(
+            vault.normalizedBorrowedAmount,
+            core.currentBorrowIndex()
+        );
+        uint256 precisionScalar = 10 ** (18 - uint256(repayAssetDecimals));
+        uint256 outstandingDebtScaled = outstandingDebt / precisionScalar;
+        uint256[] memory amounts = new uint256[](1);
+        amounts[0] = outstandingDebtScaled;
+
+        // get a flash loan of the amount of the current debt
+        bytes memory params = abi.encode(
+            _core,
+            _owner,
+            _creditScoreProof
+        );
+        address[] memory assets = new address[](1);
+        assets[0] = address(repayAsset);
+
+        uint256[] memory modes = new uint256[](1);
+        modes[0] = 0; // no debt
+
+        LENDING_POOL.flashLoan(
+            address(this),
+            assets,
+            amounts,
+            modes,
+            address(this),
+            params,
+            0
+        );
+    }
+
+    /**
+     * @dev This function is called by the Aave pool after the funds have been transferred.
+     * It executes the liquidation, then swaps the gained collateral for the stablecoin and
+     * transfers it to the original sender that called `liquidate()`.
```

Comment:
Comment is outdated now?

Reply:
crap yes it is. I'll update it but won't reupload because I don't wanna waste $60